rse-cmdline-git.Rmd

```diff
@@ -0,0 +1,2296 @@
+
+
+## Setting up Git {#rse-cmdline-git-setup}
+
+When we use Git at the command line for the first time,
+we need to configure a few things. 
+
+On a command line, Git commands are written as `git verb options`,
+where `verb` is what we actually want to do
+and `options` is additional optional information which may or may not be required for the `verb`.
+Here is how Dracula sets up his new laptop:
+
+```shell
+$ git config --global user.name "Vlad Dracula"
+$ git config --global user.email "vlad@tran.sylvan.ia"
+```
+
+(Please use your own name and email address instead of Dracula's.)
+
+For these lessons we will be interacting with [GitHub](https://github.com/),
+so the email address should be the same as the one used when setting up your GitHub account.
+
+The commands we just ran above only need to be run once: the flag `--global` tells Git
+to use the settings for every project, in your user account, on this computer.
+You can re-run them at any time if you want to change your details.
+
+You can also check your settings at any time:
+
+```shell
+$ git config --list
+```
+
+> **Git Help and Manual**
+>
+> Always remember that if you forget a `git` command,
+> you can access the list of commands by using `-h` and access the Git manual by using `--help` :
+>
+> ```shell
+> $ git config -h
+> $ git config --help
+> ```
+
+
+## Creating a repository {#rse-cmdline-git-repos}
+
+Once Git is configured,
+we can start using it.
+
+We will continue with the story of Wolfman and Dracula who are investigating if it
+is possible to send a planetary lander to Mars. 
+
+*TODO: Figure - Motivating Example [../fig/motivatingexample.png](https://github.com/swcarpentry/git-novice/blob/gh-pages/fig/motivatingexample.png)*
+
+First, let's create a directory in `Desktop` folder for our work and then move into that directory:
+
+```shell
+$ cd ~/Desktop
+$ mkdir planets
+$ cd planets
+```
+
+Then we tell Git to make `planets` a repository — a place where Git can store versions of our files:
+
+```shell
+$ git init
+```
+
+It is important to note that `git init` will create a repository that
+includes subdirectories and their files---there is no need to create
+separate repositories nested within the `planets` repository,
+whether subdirectories are present from the beginning or added later.
+Also, note that the creation of the `planets` directory and its initialization as a
+repository are completely separate processes.
+
+If we use `ls` to show the directory's contents,
+it appears that nothing has changed:
+
+```shell
+$ ls
+```
+
+But if we add the `-a` flag to show everything,
+we can see that Git has created a hidden directory within `planets` called `.git`:
+
+```shell
+$ ls -a
+```
+
+```text
+.	..	.git
+```
+
+Git uses this special sub-directory to store all the information about the project, 
+including all files and sub-directories located within the project's directory.
+If we ever delete the `.git` sub-directory,
```

Comment:
This part here could be used as a "call out/reminder" to just say, "hey, the whole history is in the .git folder"